### PR TITLE
Fixes hardcoded directory strings with env variables

### LINKED ISSUE #2639: Scenario "should push only the modified files" fails on 4.4 cluster because of No such file or directory in "/tmp/src/"
/kind bug

## What versions of software are you using?

**Operating System:**
All supported
**Output of `odo version`:**
master
## How did you run odo exactly?
make test-cmd-push

## Actual behavior
Fails

## Expected behavior
Should pass 

## Any logs, error output, etc?

```
------------------------------
odo push command tests Test push outside of the current working direcory 
  should push only the modified files
  /Users/amit/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:270

Creating a new project: wldapjazfv
Running odo with args [odo project create wldapjazfv -w -v4]
[odo]  •  Waiting for project to come up  ...
[odo] I0226 17:22:41.209372    3218 occlient.go:538] Project wldapjazfv now exists
[odo]  ✓  Waiting for project to come up [2s]
[odo]  ✓  Project 'wldapjazfv' is ready for use
[odo]  ✓  New project created and now using project: wldapjazfv
Created dir: /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467
Current working dir: /Users/amit/go/src/github.com/openshift/odo/tests/integration
Current working dir: /Users/amit/go/src/github.com/openshift/odo/tests/integration
Setting current dir to: /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467
Running git with args [git clone https://github.com/openshift/nodejs-ex /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex]
[git] Cloning into '/var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex'...
Running odo with args [odo component create nodejs:latest nodejs --project wldapjazfv --context /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex --app app]
[odo] Warning: nodejs:latest is not fully supported by odo, and it is not guaranteed to work
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [568ms]
[odo] 
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args [odo url create --port 8080 --context /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex]
[odo]  ✓  URL nodejs-8080 created for component: nodejs
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push`
Running odo with args [odo push --context /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex]
[odo] Validation
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [737ms]
[odo] 
[odo] Configuration changes
[odo]  ✓  Initializing component
[odo]  •  Creating component  ...
[odo]  ✓  Creating component [2s]
[odo] 
[odo] Applying URL changes
[odo]  ✓  URL nodejs-8080: http://nodejs-8080-app-wldapjazfv.apps.gamore44ci2602.devcluster.openshift.com created
[odo] 
[odo] Pushing to component nodejs of type local
[odo]  •  Checking files for pushing  ...
[odo]  ✓  Checking files for pushing [3ms]
[odo]  •  Waiting for component to start  ...
[odo]  ✓  Waiting for component to start [30s]
[odo]  •  Syncing files to the component  ...
[odo]  ✓  Syncing files to the component [6s]
[odo]  •  Building component  ...
[odo]  ✓  Building component [12s]
[odo]  ✓  Changes successfully pushed to component
Running oc with args [oc get route -n wldapjazfv -l app.kubernetes.io/instance=nodejs -l app.kubernetes.io/part-of=app -o jsonpath={.items[0].spec.host}]
[oc] nodejs-8080-app-wldapjazfv.apps.gamore44ci2602.devcluster.openshift.comChecking http://nodejs-8080-app-wldapjazfv.apps.gamore44ci2602.devcluster.openshift.com, for Welcome to your Node.js
try 0 of 30
Running oc with args [oc get pods --namespace wldapjazfv --selector=deploymentconfig=nodejs-app -o jsonpath='{.items[0].metadata.name}']
[oc] 'nodejs-app-1-qhf7h'Running oc with args [oc exec nodejs-app-1-qhf7h --namespace wldapjazfv -c nodejs-app -- stat /tmp/src/server.js]
[oc] stat: cannot stat '/tmp/src/server.js': No such file or directory
[oc] command terminated with exit code 1
Running oc with args [oc get pods --namespace wldapjazfv --selector=deploymentconfig=nodejs-app -o jsonpath='{.items[0].metadata.name}']
[oc] 'nodejs-app-1-qhf7h'Running oc with args [oc exec nodejs-app-1-qhf7h --namespace wldapjazfv -c nodejs-app -- stat /tmp/src/views/index.html]
[oc] stat: cannot stat '/tmp/src/views/index.html': No such file or directory
[oc] command terminated with exit code 1
Replacing "Welcome to your Node.js application on OpenShift" with "UPDATED!" in /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex/views/index.html
Running odo with args [odo push --context /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467/nodejs-ex]
[odo] Validation
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [751ms]
[odo] 
[odo] Configuration changes
[odo]  •  Retrieving component data  ...
[odo]  ✓  Retrieving component data [1s]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [1s]
[odo] 
[odo] Applying URL changes
[odo]  ✓  URL nodejs-8080 already exists
[odo] 
[odo] Pushing to component nodejs of type local
[odo]  •  Checking file changes for pushing  ...
[odo]  ✓  Checking file changes for pushing [2ms]
[odo]  •  Waiting for component to start  ...
[odo]  ✓  Waiting for component to start [254ms]
[odo]  •  Syncing files to the component  ...
[odo]  ✓  Syncing files to the component [3s]
[odo]  •  Building component  ...
[odo]  ✓  Building component [8s]
[odo]  ✓  Changes successfully pushed to component
Checking http://nodejs-8080-app-wldapjazfv.apps.gamore44ci2602.devcluster.openshift.com, for UPDATED!
try 0 of 30
Running oc with args [oc get pods --namespace wldapjazfv --selector=deploymentconfig=nodejs-app -o jsonpath='{.items[0].metadata.name}']
[oc] 'nodejs-app-1-qhf7h'Running oc with args [oc exec nodejs-app-1-qhf7h --namespace wldapjazfv -c nodejs-app -- stat /tmp/src/views/index.html]
[oc] stat: cannot stat '/tmp/src/views/index.html': No such file or directory
[oc] command terminated with exit code 1
Running oc with args [oc get pods --namespace wldapjazfv --selector=deploymentconfig=nodejs-app -o jsonpath='{.items[0].metadata.name}']
[oc] 'nodejs-app-1-qhf7h'Running oc with args [oc exec nodejs-app-1-qhf7h --namespace wldapjazfv -c nodejs-app -- stat /tmp/src/server.js]
[oc] stat: cannot stat '/tmp/src/server.js': No such file or directory
[oc] command terminated with exit code 1
Setting current dir to: /Users/amit/go/src/github.com/openshift/odo/tests/integration
Deleting project: wldapjazfv
Running odo with args [odo project delete wldapjazfv -f]
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named nodejs
[odo] This component has following urls that will be deleted with component
[odo] URL named nodejs-8080 with host nodejs-8080-app-wldapjazfv.apps.gamore44ci2602.devcluster.openshift.com having protocol http at port 8080
[odo] No services / could not get services
[odo]  •  Deleting project wldapjazfv  ...
[odo]  ✓  Deleting project wldapjazfv [13s]
[odo]  ✓  Deleted project : wldapjazfv
Deleting dir: /var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/617678467
Setting current dir to: /Users/amit/go/src/github.com/openshift/odo/tests/integration


• Failure [117.375 seconds]
odo push command tests
/Users/amit/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:14
  Test push outside of the current working direcory
  /Users/amit/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:74
    should push only the modified files [It]
    /Users/amit/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:270

    Expected
        <string>: 
    not to equal
        <string>: 

    /Users/amit/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:333
------------------------------
```

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind code-refactoring

**What does does this PR do / why we need it**:

Some of the hardcoded strings are removed from cmd_push_test.go file and replaced with env variables.

**Which issue(s) this PR fixes**:

Fixes #2639

**How to test changes / Special notes to the reviewer**:

- make test-cmd-push should pass

### CODE REVIEW

File: tests/integration/cmd_push_test.go [hunk 1]
```diff
@@ -313,7 +316,7 @@ var _ = Describe("odo push command tests", func() {
 				cmpName,
 				appName,
 				project,
-				[]string{"stat", "/tmp/src/views/index.html"},
+				[]string{"stat", filepath.Join(dir, "src/views/index.html")},
```
Comment (amitkrout) [inline tests/integration/cmd_push_test.go]: filepath.Join(dir, "src/views/index.html") -> filepath.Join(dir, "src", "views", "index.html")

Follow the same pattern for all other occurrences

**Comment by mik-dass** [comment]
@amitkrout done

**Comment by amitkrout** [comment]
> @amitkrout done

/lgtm

**Pull request merged**